src/pg/sql/11_kmeans.sql

```diff
@@ -1,21 +1,40 @@
-CREATE OR REPLACE FUNCTION  CDB_KMeans(query text, no_clusters integer,no_init integer default 20)
+-- Spatial k-means clustering
+
+CREATE OR REPLACE FUNCTION CDB_KMeans(query text, no_clusters integer, no_init integer default 20)
 RETURNS table (cartodb_id integer, cluster_no integer) as $$
-    
+
     from crankshaft.clustering import kmeans
-    return kmeans(query,no_clusters,no_init)
+    return kmeans(query, no_clusters, no_init)
+
+$$ LANGUAGE plpythonu;
 
-$$ language plpythonu;
+-- Non-spatial k-means clustering
+-- query: sql query to retrieve all the needed data
+
+CREATE OR REPLACE FUNCTION CDB_KMeansNonspatial(
+  query TEXT,
+  colnames TEXT[],
+  num_clusters INTEGER,
+  id_col TEXT DEFAULT 'cartodb_id',
+  standarize BOOLEAN DEFAULT true
+)
+RETURNS TABLE(cluster_label text, cluster_center text, rowid bigint) AS $$
```

Comment:
Done, but the output in carto isn't optimal for display:
<img width="326" alt="screen shot 2016-11-10 at 11 47 26" src="https://cloud.githubusercontent.com/assets/1041056/20185883/d3c136f0-a73b-11e6-8832-2c9b9724e825.png">

Outputting as a string will allow people to view the data in the builder, and my guess is that if people need to access it they may have the chops to know how to convert to a postgres json object and then parse it... On the other hand, we could ask for better display of json in the builder data view.